### PR TITLE
HP-1596/Add_warranty_period_for_models_and_parts

### PR DESCRIPTION
**Pull request by ValeriyShnurovoy**
## Summary by CodeRabbit

- **New Features**
    - Introduced a new `ModelColumn` class for enhanced model information display, including warranty labels.
    - Added `warranty_till` field across various parts of the application for tracking warranty expiration dates.
    - Implemented new attributes (`warranty_in_all_part` and `warranty_months`) to models for detailed warranty tracking.
    - Enhanced form views with `DateTimePicker` for `warranty_till` and fields for `warranty_months` and `warranty_in_all_part`, improving user interaction with warranty information.

### CODE REVIEW

**Review by tafid — APPROVED** [review]

File: src/grid/ModelColumn.php [hunk 1]
```diff
@@ -0,0 +1,51 @@
+<?php
+
```
Comment (SilverFire) [inline src/grid/ModelColumn.php:2]: declare strict

File: src/grid/ModelColumn.php [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    {
+        parent::init();
+
+        $this->attribute = 'model';
+        $this->prepareValue();
+    }
+    
+    private function prepareValue(): void
+    {
+        $this->value = function ($model): string {
```
Comment (SilverFire) [inline src/grid/ModelColumn.php:24]: type hint model

File: src/views/model/_form.php [hunk 3]
```diff
@@ -63,6 +63,7 @@ function getAdditionl(elem) {
         'short',
         'descr',
         'url',
+        'warranty_months',
```
Comment (SilverFire) [inline src/views/model/_form.php:66]: warranty_in_all_part field is probably missing?

File: src/models/Model.php [hunk 4]
```diff
@@ -143,6 +143,7 @@ public function attributeLabels()
             'group' => Yii::t('hipanel:stock', 'Group'),
             'group_id' => Yii::t('hipanel:stock', 'Group'),
             'is_favorite' => Yii::t('hipanel:stock', 'Is favorite'),
+            'warranty_months' => Yii::t('hipanel:stock', 'Warranty months'),
```
Comment (SilverFire) [inline src/models/Model.php:146]: warranty_in_all_part translation is missing

File: src/grid/ModelColumn.php [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+                $modelLabel = Html::a($modelLabel, ['@model/view', 'id' => $model->model_id]);
+            }
+            if (isset($model->warranty_till)) {
+                $modelLabel .= $this->getWarrantyLabel($model);
+            }
+            return $modelLabel;
+        };
+    }
+
+    private function getWarrantyLabel($model): string
```
Comment (SilverFire) [inline src/grid/ModelColumn.php:36]: type hint

File: src/grid/ModelColumn.php [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+        $color = 'info';
+        if ($diff <= 6) {
+            $color = 'warning';
+        }
+        if (!is_numeric($diff)) {
+            $color = 'danger';
+        }
+        return Label::widget(['label' => $diff, 'tag' => 'sup', 'color' => $color]);
+    }
+}
```
Comment (SilverFire) [inline src/grid/ModelColumn.php:51]: Enable newlines at the end of file in PHPStrom

**Review by SilverFire — CHANGES_REQUESTED** [review]

File: src/models/Model.php [hunk 7]
```diff
@@ -143,6 +143,8 @@ public function attributeLabels()
...
+            'set_warranty_till_in_exising_parts' => Yii::t('hipanel:stock', 'Set warranty till in exising parts'),
```
Comment (SilverFire) [inline src/models/Model.php:147]: Missing translation

File: src/models/Model.php [hunk 8]
```diff
@@ -94,8 +94,8 @@ public function rules()
                 ],
                 'safe',
             ],
-            [['is_favorite', 'show_deleted'], 'boolean'],
-            [['id', 'type_id', 'tariff_id', 'group_id'], 'integer'],
+            [['is_favorite', 'show_deleted', 'set_warranty_till_in_exising_parts'], 'boolean'],
```
Comment (SilverFire) [inline src/models/Model.php:97]: Attribute should be enabled in `create`, `update` scenarios

**Review by SilverFire — APPROVED** [review]

**Review by SilverFire — CHANGES_REQUESTED** [review]
See comments

**Review by SilverFire — APPROVED** [review]

**Pull request merged**